examples/argo.py

```diff
@@ -0,0 +1,76 @@
+import time
+
+import dgl.multiprocessing as dmp
+import psutil
+from skopt import gp_minimize
+
+
+class ARGO:
+    def __init__(self, n_search=10, epoch=200, batch_size=4096,
+                 space=[(2, 8), (1, 4), (1, 32)], random_state=1):
+        self.n_search = n_search # number of configuration searches the auto-tuner will conduct
+        self.epoch = epoch # number of epochs of training
+        self.batch_size = batch_size # size of the mini-batch
+        self.space = space # range of the search space, [number of processes, number of samplers, number of trainers]
+        self.random_state = random_state # number of random initializations before searching
+        self.acq_func = 'EI' # acqusition function of the auto-tuner
+        self.counter = [0]
+
+    def core_binder(self, num_cpu_proc, n_samp, n_train, rank):
+        load_core, comp_core = [], []
+        n = psutil.cpu_count(logical=False)
+        size = num_cpu_proc
+        num_of_samplers = n_samp
+        load_core = list(
+            range(n // size * rank, n // size * rank + num_of_samplers))
+        comp_core = list(
+            range(n // size * rank + num_of_samplers,
+                  n // size * rank + num_of_samplers + n_train))
+        return load_core, comp_core
+
+    def auto_tuning(self, train, args):
+        ep = 1
+        result = gp_minimize(lambda x: self.mp_engine(x, train, args, ep),
+                             dimensions=self.space, n_calls=self.n_search,
+                             random_state=self.random_state,
+                             acq_func=self.acq_func)
+        return result
+
+    def mp_engine(self, x, train, args, ep):  # Multi-Process Engine
```

Comment:
Type hints here too.

Reply:
Thanks for your suggestions!
I added type hints to each function and provided a short doc.
The description of ARGO is available on top of the file now.
For benchmarking results, they are available in the paper (Fig. 10 and 11).
Please let me know if you need anything else from me :)